### PR TITLE
Fix submission check in LHCbTransform

### PR DESCRIPTION
**Pull request by chrisburr**
When `files_per_unit <= 0`, `LHCbUnit.input_datset_index` isn't set causing ganga to continuously create new identical units of work.

I've also removed the duplication between the split/unsplit case that (presumably) caused this bug to sneak in.

### CODE REVIEW

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
test this please

**Comment by mesmith75** [comment]
test this please

**Comment by mesmith75** [comment]
retest this please

File: python/GangaLHCb/Lib/Tasks/LHCbTransform.py [hunk 1]
```diff
@@ -117,38 +117,26 @@ def createUnits(self):
...
-                        unit.inputdata.files += new_data.files[
-                            num:num + self.files_per_unit]
-                        num += self.files_per_unit
-
-                else:
-                    # new unit required for this dataset
+                # Create units for these files
+                step = self.files_per_unit
+                if step <= 0:
+                    step = int(1e100)
```
Comment (mesmith75) [inline python/GangaLHCb/Lib/Tasks/LHCbTransform.py]: I'm not sure I like setting this to an arbitrarily large size. Would it not be better to set it to `len(new_data.files)`?

Comment (chrisburr) [inline python/GangaLHCb/Lib/Tasks/LHCbTransform.py]: I'm not sure what I was thinking to leave it like that, I've changed it.

**Comment by mesmith75** [comment]
test this please

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
test this please

**Comment by mesmith75** [comment]
Does anyone know how to get the travis test to run?

**Comment by egede** [comment]
@mesmith75 Given that the Jenkins testing runs the same tests, I would probably just ignore that the travis tests are not starting and merge using the command line options.

**Comment by mesmith75** [comment]
@egede Sure, although here it looks as if something is going awry with the PR quality tests.

**Comment by mesmith75** [comment]
restest this please

**Comment by mesmith75** [comment]
Thanks @chrisburr

**Pull request merged**